src/pages/wishlist/WishlistCard.jsx

```diff
@@ -1,22 +1,55 @@
 import wishlist from './Wishlist.module.css'
-import { image } from '../../assets/images/index'
+import { useWishlistContext } from '../../context/index'
 
-export function WishlistCard() {
+export function WishlistCard(product) {
+  const { wishlistItems, setWishlistItems } = useWishlistContext()
+
+  const { _id, title, img, price, oldPrice, discount, categoryName, rating } =
+    product
+
+  function addWishlistItem() {
+    for (let i = 0; i < wishlistItems.length; i++) {
+      if (product._id === wishlistItems[i]._id)
+        return setWishlistItems(
+          wishlistItems.filter(({ _id }) => _id !== product._id)
+        )
+    }
+    setWishlistItems([...wishlistItems, product])
+  }
+  function containsInWishlist(_id) {
+    return wishlistItems.some(product => product._id === _id)
+  }
```

Comment:
Please lift up this to wishlist context. It will avoid repetitive code.

Reply:
Thanks for suggesting, I have now put this functions into utils folder. Please review and approve the pull request